### PR TITLE
Fix codestarts encoding problem

### PR DESCRIPTION
**Pull request by ia3andy**
Try to fix encoding problem as reported by @gastaldi:
Also there are some encoding errors in the produced file:
```
Be aware that it?s not an _?ber-jar_ as the dependencies are copied into the `target/lib` directory.
If you want to build an _?ber-jar_, just add the `--uber-jar` option to the command line:
```

### CODE REVIEW

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/CodestartFileStrategyHandler.java [hunk 1]
```diff
@@ -48,4 +49,8 @@ default void createDirectories(Path targetPath) throws IOException {
         Files.createDirectories(targetPath.getParent());
     }
 
+    default void writeFile(final Path targetPath, final String content) throws IOException {
+        Files.write(targetPath, content.getBytes(StandardCharsets.UTF_8));
```
Comment (gastaldi) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/CodestartFileStrategyHandler.java:53]: This won't work for binary files

Comment (ia3andy) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/CodestartFileStrategyHandler.java:53]: @gastaldi I don't use String for binary files :)

Binary files are directly copied: 
https://github.com/quarkusio/quarkus/blob/master/independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartResource.java#L145

Comment (gastaldi) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/CodestartFileStrategyHandler.java:53]: Ah doh, I missed the method signature 😝 

LGTM then

**Review by gastaldi — APPROVED** [review]

**Comment by gsmet** [comment]
Just to be sure, did someone check that the aforementioned encoding issues are fixed?

It's not clear from the description or @gastaldi 's comment.

Thanks.

**Comment by gastaldi** [comment]
That was happening in code.quarkus.io, which @ia3andy fixed in https://github.com/quarkusio/code.quarkus.io/commit/bc0734a357dc2ad5f2d87acd575454ed6223ca47.

But as @ia3andy mentioned in Zulip, UTF-8 may not be the default encoding when running the codestart generation, hence this PR

**Comment by ia3andy** [comment]
@gsmet yes I have fixed it on code.quarkus.io using a workaround but the result is the same (until 1.10): https://github.com/quarkusio/code.quarkus.io/commit/bc0734a357dc2ad5f2d87acd575454ed6223ca47

**Pull request merged**